File: app/src/main/java/org/kindone/willingtodo/TaskEditActivity.java
package org.kindone.willingtodo;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.util.Log;
import android.view.Menu;
import android.view.MenuItem;
import android.view.inputmethod.InputMethodManager;
import android.widget.EditText;

public class TaskEditActivity extends TaskFormActivity {

    private static String ARG_TASK_ID = MainActivity.RESULT_TASK_ID;

    private long mTaskId;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        Intent intent = getIntent();
        long taskId = intent.getLongExtra(ARG_TASK_ID, -1);
        String taskTitle = intent.getStringExtra(ARG_TASK_TITLE);

        initializeView(taskTitle);
        initializeTaskId(taskId);
    }


    @Override
    protected boolean processOptionItemSelected(MenuItem item)
    {
        int id = item.getItemId();

        switch(id)
        {
            case R.id.action_done:
                setActivityResult(RESULT_OK, mTaskId, getTaskTitle());
                finish();
                return true;
            default:
                return false;
        }
    }


    private void setActivityResult(int status, long taskId, String taskTitle)
    {
        Intent intent = this.getIntent();
        intent.putExtra(ARG_TASK_ID, taskId);
        intent.putExtra(ARG_TASK_TITLE, taskTitle);

        if (getParent() == null) {
            setResult(status, intent);
        } else {
            getParent().setResult(status, intent);
        }
    }


    private void initializeTaskId(long id)
    {
        mTaskId = id;
    }




}

File: app/src/main/java/org/kindone/willingtodo/TaskEditActivity.kt
package org.kindone.willingtodo

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar
import android.util.Log
import android.view.Menu
import android.view.MenuItem
import android.view.inputmethod.InputMethodManager
import android.widget.EditText

class TaskEditActivity : TaskFormActivity() {

    private var mTaskId: Long = 0


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val intent = intent
        val taskId = intent.getLongExtra(ARG_TASK_ID, -1)
        val taskTitle = intent.getStringExtra(TaskFormActivity.Companion.ARG_TASK_TITLE)

        initializeView(taskTitle)
        initializeTaskId(taskId)
    }


    override fun processOptionItemSelected(item: MenuItem): Boolean {
        val id = item.itemId

        when (id) {
            R.id.action_done -> {
                setActivityResult(RESULT_OK, mTaskId, taskTitle)
                finish()
                return true
            }
            else -> return false
        }
    }


    private fun setActivityResult(status: Int, taskId: Long, taskTitle: String) {
        val intent = this.intent
        intent.putExtra(ARG_TASK_ID, taskId)
        intent.putExtra(TaskFormActivity.Companion.ARG_TASK_TITLE, taskTitle)

        if (parent == null) {
            setResult(status, intent)
        } else {
            parent.setResult(status, intent)
        }
    }


    private fun initializeTaskId(id: Long) {
        mTaskId = id
    }

    companion object {

        private val ARG_TASK_ID = MainActivity.RESULT_TASK_ID
    }


}
